[Flang] Move genMinMaxlocReductionLoop to Transforms/Utils.cpp

davemgreen opened the pull request:
This is one option for attempting to move genMinMaxlocReductionLoop to a better location. It moves it into Transforms and makes HLFIRTranforms depend upon FIRTransforms.

It passes a build locally, both with and without -DBUILD_SHARED_LIBS, but that might not include all configurations.

davemgreen commented:
This looks like it does OK locally and in CI. Let me know if you think there is a better place for it.

vzakhari reviewed (approved):
Thank you, @davemgreen.  It seems okay that `HLFIRTransforms` depends on `FIRTransforms`.

davemgreen committed:
[Flang] Move genMinMaxlocReductionLoop to Transforms/Utils.cpp

This is one option for attempting to move genMinMaxlocReductionLoop to a better
location. It moves it into Transforms and makes HLFIRTranforms depend upon
FIRTransforms.

davemgreen commented:
Thanks. Let me know if this causes problems in some other configuration.
